pkg/oam/util/helper.go

```diff
@@ -235,7 +250,24 @@ func GetCapabilityDefinition(ctx context.Context, cli client.Reader, definition
 	return nil
 }
 
-func fetchDefinitionRev(ctx context.Context, cli client.Reader, definitionName string) (bool, *v1beta1.DefinitionRevision, error) {
+func getDefinitionType(definition client.Object) (common.DefinitionType, error) {
+	var definitionType common.DefinitionType
+	switch definition.(type) {
+	case *v1beta1.ComponentDefinition:
+		definitionType = common.ComponentType
+	case *v1beta1.TraitDefinition:
+		definitionType = common.TraitType
+	case *v1beta1.PolicyDefinition:
+		definitionType = common.PolicyType
+	case *v1beta1.WorkflowStepDefinition:
+		definitionType = common.WorkflowStepType
+	default:
+		return definitionType, fmt.Errorf("invalid definition")
+	}
+	return definitionType, nil
+}
+
+func fetchDefinitionRev(ctx context.Context, cli client.Reader, definitionName string, definitionType common.DefinitionType, fctx map[string]string) (bool, *v1beta1.DefinitionRevision, error) {
```

Comment:
I don't think we need a map here, we can just pass the parameter `autoUpdate bool`.